File: packages/tactical-engine/src/targeting/resolve-cylinder.ts
import type { Coordinate, EdgeObstacle, Obstacle } from "@limiarmap/shared-contracts";
import { resolveSphere } from "./resolve-sphere";

/**
 * Resolves a cylinder-shaped AoE on the 2D tactical grid.
 *
 * Cylinder height is intentionally ignored by LimiarMap: combat targeting uses
 * the cylinder's ground projection only. Its footprint is therefore the same
 * circular Chebyshev-radius approximation used by sphere, centered on the
 * selected anchor cell, including the same line-of-effect filtering.
 */
export function resolveCylinder(
  center: Coordinate,
  radius: number,
  obstacles: Obstacle[],
  edgeObstacles: EdgeObstacle[] = []
): Coordinate[] {
  return resolveSphere(center, radius, obstacles, edgeObstacles);
}

File: packages/tactical-engine-py/tactical_engine/targeting/resolve_cylinder.py
from shared_contracts import Coordinate, Obstacle
from .resolve_sphere import resolve_sphere


def resolve_cylinder(
    center: Coordinate,
    radius: int,
    obstacles: list[Obstacle],
) -> list[Coordinate]:
    """Resolve a cylinder AoE as a 2D circular footprint.

    LimiarMap does not model vertical height in combat targeting. Cylinder
    height is ignored, and the selected point is treated as the center of the
    cylinder's ground projection using the same Chebyshev approximation as
    sphere.
    """
    return resolve_sphere(center, radius, obstacles)
